crates/re_types_builder/src/codegen/cpp/mod.rs

```diff
@@ -812,36 +812,47 @@ impl QuotedObject {
         };
 
         let copy_constructor = {
-            let copy_match_arms = obj
-                .fields
-                .iter()
-                .filter_map(|obj_field| {
-                    // Inferring from trivial destructability that we don't need to call the copy constructor is a little bit wonky,
-                    // but is typically the reason why we need to do this in the first place - if we'd always memcpy we'd get double-free errors.
-                    // (As with swap, we generously assume that objects are rellocatable)
-                    (!obj_field.typ.has_default_destructor(objects)).then(|| {
-                        let tag_ident = format_ident!("{}", obj_field.name);
-                        let field_ident =
-                            format_ident!("{}", crate::to_snake_case(&obj_field.name));
-                        Some(quote! {
-                            case detail::#tag_typename::#tag_ident: {
-                                _data.#field_ident = other._data.#field_ident;
-                                break;
-                            }
-                        })
-                    })
-                })
-                .collect_vec();
+            // Note that `switch` on an enum without handling all cases causes `-Wswitch-enum` warning!
+            let mut copy_match_arms = Vec::new();
+            let mut default_match_arms = Vec::new();
+            for obj_field in &obj.fields {
+                let tag_ident = format_ident!("{}", obj_field.name);
+                let case = quote!(case detail::#tag_typename::#tag_ident:);
+
+                // Inferring from trivial destructability that we don't need to call the copy constructor is a little bit wonky,
+                // but is typically the reason why we need to do this in the first place - if we'd always memcpy we'd get double-free errors.
+                // (As with swap, we generously assume that objects are rellocatable)
+                if obj_field.typ.has_default_destructor(objects) {
+                    default_match_arms.push(case);
+                } else {
+                    let field_ident = format_ident!("{}", crate::to_snake_case(&obj_field.name));
+                    copy_match_arms.push(quote! {
+                        #case {
+                            _data.#field_ident = other._data.#field_ident;
+                            break;
+                        }
+                    });
+                }
+            }
+
+            let trivial_memcpy = quote! {
+                const void *otherbytes = reinterpret_cast<const void *>(&other._data);
+                void *thisbytes = reinterpret_cast<void *>(&this->_data);
```

Comment:
nit: let use the same style as we specify in `.clang-format`

```suggestion
                const void* otherbytes = reinterpret_cast<const void*>(&other._data);
                void* thisbytes = reinterpret_cast<void *>(&this->_data);
```